modules/common/vehicle_state/vehicle_state.cc

```diff
@@ -27,7 +27,7 @@ namespace vehicle_state {
 VehicleState::VehicleState(
     const localization::LocalizationEstimate &localization) {
   ConstructExceptLinearVelocity(&localization);
-  if (localization.has_pose() && localization.pose().has_linear_velocity()) {
+  if (localization != nullptr && localization.has_pose() && localization.pose().has_linear_velocity()) {
```

Comment:
localization is a reference, it won't point to a nullptr.
Have you tried to compile your code?